src/main/kotlin/ru/tinkoff/fintech/refactoring/Employee.kt

```
@@ -1,72 +1,61 @@
 package ru.tinkoff.fintech.refactoring
 
-interface Employee {
-    fun makePizza(orderId: Int, pizza: Pizza, ingredients: List<Pair<String, Int>>)
-    fun makeCoffee(orderId: Int, coffee: Coffee)
-    fun cleanFloor()
+interface Employee {              // open-closed principle
+
+    fun workInfo(): String
+
 }
 
-class Barista : Employee {
-    override fun makePizza(orderId: Int, pizza: Pizza, ingredients: List<Pair<String, Int>>) {
-        println("[Бариста] Я не умею готовить пиццу")
-    }
+class Barista(private val coffeeOrder: CoffeeOrder) : Employee {
 
-    override fun cleanFloor() {
-        println("[Бариста] Я не умею мыть полы")
-    }
+    private val roundedPrice = "%.2f".format(coffeeOrder.getPriceCoffee())
+
+    private var textWorkExecute = listOf(
+        "[Бариста] Готовлю напиток: ${coffeeOrder.getNameCoffee()}",
+        "[Бариста] Время приготовления: ${coffeeOrder.getTimeInMinutesCoffee()} минут",
+        "[Бариста] Стоимость напитка: $roundedPrice$",
+        "[Бариста] Заказ ${coffeeOrder.getNumberOrder()} готов\n").joinToString("\n")
 
-    override fun makeCoffee(orderId: Int, coffee: Coffee) {
-        println("[Бариста] Готовлю напиток: ${coffee.name}")
-        println("[Бариста] Время приготовления: ${calculateCoffeeBrewTimeInMinutes(coffee)} минут")
-        val roundedPrice = "%.2f".format(coffee.price)
-        println("[Бариста] Стоимость напитка: $roundedPrice")
+    private var textWorkFail = listOf(
+        "[Бариста] Какой экзотический кофе: ${coffeeOrder.getNameCoffee()}!",
+        "[Бариста] Заказ ${coffeeOrder.getNumberOrder()} отклонен\n").joinToString("\n")
 
-        println("[Бариста] заказ $orderId готов")
+    override fun workInfo() = when (coffeeOrder.getTimeInMinutesCoffee()) {
+        0 -> textWorkFail
```

ну вот мне кажется это не совсем корректно

если в мапе нету элемента с таким именем, то мы генерим пустышку с нулями, а если время приготовления равно нулю, но говорим что кофе слишком экзотический

если прочитать это три раза, то вроде бы и понятна схема. но желательно чтобы она была понятна с первого раза

кстати, если заказ заведомо невыполним, может тогда не стоит передавать его баристе? можно падать (fail fast) еще на формировании заказа. было бы другое дело, если бы заказ был теоретически выполним, но у баристы был бы свой список того, что он умеет делать, и там не было бы этого экзотического кофе

---

понял. падаю